tests/analyses/test_polarization.py

```diff
@@ -1,8 +1,49 @@
 import numpy as np
 import pytest
 
-import zospy as zp
-from zospy.analyses.polarization import polarization_pupil_map, transmission
+import zospy.api.config as _config
+from zospy.analyses.polarization import (
+    _get_number_field,
+    polarization_pupil_map,
+    transmission,
+)
+
+_signs = ["", "+", "-"]
+_int_numbers = ["1", "123"]
+_float_numbers = [".1", ".123", "1.", "1.2", "1.23", "12.3"]
+_decimal_separators = [",", "."]
+_exps = ["", "e1", "e123", "e+1", "e+123", "e-1", "e-123", "E1", "E123", "E+1", "E+123", "E-1", "E-123"]
+
+
+class TestGetNumberField:
+    @pytest.mark.parametrize(
```

Comment:
It is possible to add multiple paramterize decorators to a single test, which effectively generates a test matrix and looks cleaner.

```suggestion
    @pytest.mark.parametrize("sign", _signs)
    @pytest.mark.parametrize("number", _int_numbers)
    @pytest.mark.parametrize("exp", _exps)
    def test_get_number_field_returns_correct_result_for_integers(self, sign, number, exp):
        ...
```